library/entropy.c

```diff
@@ -292,8 +299,14 @@ static int entropy_gather_internal( mbedtls_entropy_context *ctx )
         }
     }
 
-    if( have_one_strong == 0 )
-        ret = MBEDTLS_ERR_ENTROPY_NO_STRONG_SOURCE;
+    if( have_one_strong_fi == 0 )
```

Comment:
Do we need to have double check for the failing branch?

Can we have positive check, something like "if control_flow OK and `have_one_strong_fi` then return "ret" and by default return MBEDTLS_ERR_ENTROPY_NO_STRONG_SOURCE (or FAULT)"